### ISSUE TITLE
Access view-items LookupField

### ISSUE THREAD

**Issue by tobiashoeft** [SharePoint/PnP-JS-Core#624]
#### Category
[ ] Enhancement

[ ] Bug

[X] Question
####
Hi is it possible to access  view the view items and then get the value from a lookupfield? like in CSOM (LookUpFieldValue)item["bla"]  ?

**Comment by koltyakov** [comment]
Hey Tobias,

If I understood you correctly you need to expand lookups field's Title. It can easily be done with $select and $expand modifiers like this:

```javascript
pnp.sp.web.getList('/sites/site/Lists/ListEntityName')
  .items
  .select('LookupField/Id,LookupField/Title')
  .expand('LookupField')
  .get()
  .then(console.log)
```

**Comment by tobiashoeft** [comment]
Sorry the writting getting faster then the brain ;).

Yeah that way i know, but can i select an SPListView and select in this view the LookUpField? 
I tried it but i dont get an working example

**Comment by koltyakov** [comment]
Well...
Let me rephrase it, please. 
So you need:
- Get a list's view fields
- And understand which of them are lookups
Do you?

```javascript
let list = pnp.sp.web.getList('/sites/site/Lists/ListEntityName');
Promise.all([
  list.views.getByTitle('All Items').fields.get(),
  list.fields.filter(`TypeAsString eq 'Lookup'`).get()
])
  .then(([viewFields, lookupFields]) => {
    return lookupFields.filter(field => {
      return viewFields.Items.indexOf(field.InternalName) !== -1;
    });
  })
  .then(lookupFieldsInAView => console.log(lookupFieldsInAView))
  .catch(console.log);
```

**Comment by tobiashoeft** [comment]
Thx you for reply!

Thanks for the reply but I would like something else.
But the example of you shows me further.
Again to my issue
I want to display the items from a list with a specific view.

In this list are "LookupFields" these lookupfields I would like to represent synonymous with and I wanted to ask whether one from this particular view synonymous to the lookupfelder Values to grasp, I find unfortunately nothing in the documentation.
 
Here in CSOM for example backend ( api ) 
```csharp
            List list = context.Web.Lists.GetByTitle(listName);
            context.Load(list );
            await Task.Run(() => context.ExecuteQuery());
            View view = list.Views.GetByTitle(listView);
            context.Load(view);
            await Task.Run(() => context.ExecuteQuery());
            CamlQuery query = new CamlQuery
            {
                ViewXml = view.ViewQuery
            };
            ListItemCollection items = list .GetItems(query);
            context.Load(items);
            await Task.Run(() => context.ExecuteQuery());


            foreach (ListItem item in items)
            {
                if (!item[valueSource].ToString().Equals(valueForgein)) continue;
                value = GetValueFromLookUpField(item, returnValue);
                break;
            }

            return value;
```

**Comment by koltyakov** [comment]
It's kind of tricky to take view query and embrace it with OData request flawlessly.
There are `ListViewXml` and `ViewQuery` properties in a view data response which can be used.
There is `getItemsByCAMLQuery` list's method but it was implemented in REST for backward compatibility and it is not OData so can't be chained with OData modifiers ($select, $expand and so on) and rarely recommended to use.
I would recommend stepping aside from CSOM implementation a bit for more native REST approach.
Or use JSOM for this, maybe, which logic can be projected from existing CSOM one.
Or maybe I am not getting still what you're trying to achieve this Friday night. =) Sounds like lookup values localization. Giving up for today. Luckily we're not alone here.

**Comment by tobiashoeft** [comment]
:D Maybe it's to late for me to explain these issue deeper but i think you got it, what i mean. So maybe other guys here can think about it or get help me. Thx for all for the moment cheers sir ;)

**Comment by patrick-rodgers** [comment]
@tobiashoeft - back from family leave and catching up. I confess I am not sure what you are trying to achieve. @koltyakov has outlined several methods for working with lookup fields above. Please let us know if you are still stuck on this and if you can provide perhaps a bit more detail? Thanks!
